.github/scripts/test.sh

```diff
@@ -40,12 +40,15 @@ fi
 cd /home
 sherpa_smoke ${smokevars} || exit 1
 
+# Set command to run doctests if XSPEC and test data are present
+if [ -n "${XSPECVER}" ] && [ ${TEST} != none ]; then DOCTEST="--doctest-rst"; fi
+
 # Run regression tests using sherpa_test
 if [ ${TEST} == package ] || [ ${TEST} == none ]; then
     cd $HOME;
     conda install -yq pytest-cov;
     
     # This automatically picks up the sherpatest module when TEST==package
-    sherpa_test --cov sherpa --cov-report xml || exit 1;
+    sherpa_test $DOCTEST --cov sherpa --cov-report xml || exit 1;
```

Comment:
@DougBurke Can I ask for your help? For simplicity, I don't want to run the doctests every time, it's simpler if we run them only when both XSPEC and the TESTDATA are present. In that case, I want to pass the "--doctest-rst" to pytest. On the command line, that's simple: `pytest --doctest-rst` will be sufficient to run the doctests in the modules that I have not blacklisted. However, how do I do that in CI? We seem to have XSPEC only in conda installations and in those we don't run pytest directly, we run `sherpa_test` instead. Is there an easy way to apss through the argument?

Alternatively, I could skip anything that uses XSPEC models, but it would be nice to test those parts, too. See files changes in this PR - I discovered an example where our naming was incocnsistent already.

Reply:
So, I've spent a lot of time in #1329 / #1352 to try and modernize our CI code, where by "modernize" I really mean "clean up as we no-longer need to follow the Travis setup". From that work I can claim that I do not understand the naming scheme our tests use (e.g. the `TEST == package` vs ... options) and there do seem to be some ways you can set up the tests to be less-than-optimal (e.g. see #1504).

This is all to say that I hope our CI runs will be changed significantly soon, but this means awkwardness for any code that wants to change the behavior in the mean time, and requires the existing PRs to be reviewed.

One thing that may help in the short term is that `sherpa_test` is essentially a wrapper around `pytest` - see https://github.com/sherpa/sherpa/blob/feffe3565e329162b1dff81e34d6e3a64ab3fe9b/sherpa/__init__.py#L871-L883 - and in #1001 I added support to allow arguments to be passed through (which is why you can add `--cov ...` as shown above, if memory serves), but there are parts of `pytest` that I don't understand so although I'd hope you can just pass through the argument, I can't guarantee it'll work.

For background: we only have XSPEC available in conda because trying to get XSPEC-enabled pip installs would, I predict, be too painful. With the conda build we can pull in our test build of the XSPEC model library via conda and our channel, but for pip I don't know what you'd do. I have spent time looking at separating out the low-level interface to XSPEC models, which could then be a dependency of Sherpa, but my attempts to get HEACIT people to look at this have been less-than successful and we are no well-beyond the scope of this PR.